KeyError 45058

![image](https://user-images.githubusercontent.com/1022757/52214465-bc459300-28a2-11e9-8686-7e98247d27ab.png)


```
2019-02-04 17:28:49,695 [Analytics] [ERROR]  handle_analytic_task Exception: 'Traceback (most recent call last):
  File "C:\Projects\hastic-server\analytics\bin\..\analytics\analytic_unit_manager.py", line 94, in handle_analytic_task
    result_payload = await self.__handle_analytic_task(task)
  File "C:\Projects\hastic-server\analytics\bin\..\analytics\analytic_unit_manager.py", line 88, in __handle_analytic_task
    return await worker.do_detect(data, payload['cache'])
  File "C:\Projects\hastic-server\analytics\bin\..\analytics\analytic_unit_worker.py", line 34, in do_detect
    return self._detector.detect(data, cache)
  File "C:\Projects\hastic-server\analytics\bin\..\analytics\detectors\pattern_detector.py", line 53, in detect
    detected = self.model.detect(dataframe, cache)
  File "C:\Projects\hastic-server\analytics\bin\..\analytics\models\model.py", line 79, in detect
    ) for x in result]
  File "C:\Projects\hastic-server\analytics\bin\..\analytics\models\model.py", line 79, in <listcomp>
    ) for x in result]
  File "C:\Users\gmpot\AppData\Local\Programs\Python\Python36-32\lib\site-packages\pandas\core\series.py", line 601, in __getitem__
    result = self.index.get_value(self, key)
  File "C:\Users\gmpot\AppData\Local\Programs\Python\Python36-32\lib\site-packages\pandas\core\indexes\base.py", line 2477, in get_value
    tz=getattr(series.dtype, 'tz', None))
  File "pandas\_libs\index.pyx", line 98, in pandas._libs.index.IndexEngine.get_value (pandas\_libs\index.c:4404)
  File "pandas\_libs\index.pyx", line 106, in pandas._libs.index.IndexEngine.get_value (pandas\_libs\index.c:4087)
  File "pandas\_libs\index.pyx", line 154, in pandas._libs.index.IndexEngine.get_loc (pandas\_libs\index.c:5126)
  File "pandas\_libs\hashtable_class_helper.pxi", line 759, in pandas._libs.hashtable.Int64HashTable.get_item (pandas\_libs\hashtable.c:14031)
  File "pandas\_libs\hashtable_class_helper.pxi", line 765, in pandas._libs.hashtable.Int64HashTable.get_item (pandas\_libs\hashtable.c:13975)
KeyError: 45058
'
```

![image](https://user-images.githubusercontent.com/1022757/52214437-af28a400-28a2-11e9-9af0-f86c13da8b3b.png)

I were just labeling things. It is dev version.

--- a/analytics/analytics/models/general_model.py
+++ b/analytics/analytics/models/general_model.py
@@ -66,6 +66,7 @@ class GeneralModel(Model):
         if pat_data.count(0) == len(pat_data):
             raise ValueError('Labeled patterns must not be empty')
 
+        self.all_conv = []
         for i in range(self.state['WINDOW_SIZE'] * 2, len(data)):
             watch_data = data[i - self.state['WINDOW_SIZE'] * 2: i]
             watch_data = utils.subtract_min_without_nan(watch_data)

--- a/analytics/tests/test_dataset.py
+++ b/analytics/tests/test_dataset.py
@@ -172,14 +172,14 @@ class TestDataset(unittest.TestCase):
         data_val = [1.0, 2.0, 5.0, 2.0, 1.0, 1.0, 3.0, 6.0, 4.0, 2.0, 1.0, 0, 0]
         dataframe = create_dataframe(data_val)
         segments = [{'_id': 'Esl7uetLhx4lCqHa', 'analyticUnitId': 'opnICRJwOmwBELK8', 'from': 1523889000001, 'to': 1523889000003, 'labeled': True, 'deleted': False}]
-        try:
-            model = models.GeneralModel()
-            model_name = model.__class__.__name__
-            for _ in range(2):
-                model.fit(dataframe, segments, dict())
-                model.do_detect(dataframe)
-        except ValueError:
-            self.fail('Model {} raised unexpectedly'.format(model_name))
+        model = models.GeneralModel()
+        model_name = model.__class__.__name__
+        model.fit(dataframe, segments, dict())
+        result = len(data_val) + 1
+        for _ in range(2):
+            model.do_detect(dataframe)
+            max_pattern_index = max(model.do_detect(dataframe))
+            self.assertLessEqual(max_pattern_index, result)
 
 
 if __name__ == '__main__':

--- a/analytics/tests/test_dataset.py
+++ b/analytics/tests/test_dataset.py
@@ -167,6 +167,19 @@ class TestDataset(unittest.TestCase):
                 model.fit(dataframe, segments, dict())
         except ValueError:
             self.fail('Model {} raised unexpectedly'.format(model_name))
+    def test_general_for_two_labeling(self):
+        data_val = [1.0, 2.0, 5.0, 2.0, 1.0, 1.0, 3.0, 6.0, 4.0, 2.0, 1.0, 0, 0]
+        dataframe = create_dataframe(data_val)
+        segments = [{'_id': 'Esl7uetLhx4lCqHa', 'analyticUnitId': 'opnICRJwOmwBELK8', 'from': 1523889000001, 'to': 1523889000003, 'labeled': True, 'deleted': False}]
+        try:
+            model = models.GeneralModel()
+            model_name = model.__class__.__name__
+            for _ in range(2):
+                model.fit(dataframe, segments, dict())
+                model.do_detect(dataframe)
+        except ValueError:
+            self.fail('Model {} raised unexpectedly'.format(model_name))
+
 
 if __name__ == '__main__':
     unittest.main()

--- a/analytics/tests/test_dataset.py
+++ b/analytics/tests/test_dataset.py
@@ -167,6 +167,7 @@ class TestDataset(unittest.TestCase):
                 model.fit(dataframe, segments, dict())
         except ValueError:
             self.fail('Model {} raised unexpectedly'.format(model_name))
+
     def test_general_for_two_labeling(self):
         data_val = [1.0, 2.0, 5.0, 2.0, 1.0, 1.0, 3.0, 6.0, 4.0, 2.0, 1.0, 0, 0]
         dataframe = create_dataframe(data_val)

--- a/analytics/tests/test_dataset.py
+++ b/analytics/tests/test_dataset.py
@@ -173,7 +173,6 @@ class TestDataset(unittest.TestCase):
         dataframe = create_dataframe(data_val)
         segments = [{'_id': 'Esl7uetLhx4lCqHa', 'analyticUnitId': 'opnICRJwOmwBELK8', 'from': 1523889000001, 'to': 1523889000003, 'labeled': True, 'deleted': False}]
         model = models.GeneralModel()
-        model_name = model.__class__.__name__
         model.fit(dataframe, segments, dict())
         result = len(data_val) + 1
         for _ in range(2):

